examples/blogs__parcel_preprocessor/cypress/plugins/index.js

```diff
@@ -0,0 +1,88 @@
+const ParcelBundler = require('parcel-bundler')
+const path = require('path')
+
+/* global Promise */
+/* eslint-disable no-console */
+
+const bundleOnce = (filePath, outputPath) => {
+  const outDir = path.dirname(outputPath)
+  const outFile = path.basename(outputPath)
+
+  // seems parcel-bundler 1.7.0 does not want extension
+  // in the output filename
+  const basename = path.basename(outFile, path.extname(outFile))
+  const options = {
+    watch: false,
+    hmr: false,
+    outFile: basename,
+    outDir,
+  }
+  const bundler = new ParcelBundler(filePath, options)
+  return bundler.bundle()
+}
+
+const bundlers = {}
+
+const onFile = (file) => {
+  const { filePath, shouldWatch, outputPath } = file
+  if (!shouldWatch) {
+    console.log('bundle file once without watching to %s', outputPath)
+    return bundleOnce(filePath, outputPath).then(() => outputPath)
+  }
+
+  if (bundlers[filePath]) {
+    return bundlers[filePath]
+  }
+
+  console.log('watch and bundle file %s', outputPath)
+
+  // instead of temp folder for output files, use default "dist" folder
+  // this way the source maps are created and served next to the bundle
+
+  const options = {
+    watch: shouldWatch,
+    // make output simpler and avoid possible conflicts with Cypress
+    // by NOT having hot module reloading
+    hmr: false,
+    logLevel: 3,
+  }
+
+  const bundler = new ParcelBundler(filePath, options)
+
+  bundlers[filePath] = new Promise((resolve, reject) => {
+    bundler.bundle()
+    .then((r) => {
+      if (r) {
+        console.log('finished bundle', r.name)
+        resolve(r.name)
+      } else {
+        reject(new Error(`Could not bundle ${filePath}`))
+      }
+    })
+    .catch(reject)
+  })
+
+  bundler.on('bundled', (b) => {
```

Comment:
It's an unlikely circumstance, but this could result in an outdated file being served. Ideally there should be an event that happens when a 'rebundle' is triggered, which is where you should overwrite `bundlers[filePath]` with a new promise and then resolve that promise with this `bundled` event. So it would look like: